### ISSUE TITLE
Prisma Stopping koajs server

### ISSUE THREAD

**Issue by Daggy1234** [prisma/prisma#6602]
For some reason koa works fine when I do not initialise the prisma client, but whenever I do, it breaks.

Here is a minimal reproducible example

Does not Work.

```ts
import Koa, { Context } from 'koa';
import Router from 'koa-router';
import { PrismaClient } from '@prisma/client';


const HOST = "127.0.0.1";
const PORT = process.env.PORT || 7000;

const router = new Router();
export const app = new Koa();
router.get('/', (ctx) => ctx.body = { message: 'Welcome to MikroORM Koa TS example, try CRUD on /author and /book endpoints!' });


(async () => {
    const prisma = new PrismaClient();
    app.use(router.routes())
    app.use(router.allowedMethods())
    console.log("Added Routes");
    app.use((ctx, next) => {
      ctx.status = 404;
      ctx.body = { message: 'No route found' };
    });

    app.on("error", err => {
      console.log(err);
    })

    app.listen(PORT, () => {
      console.log("Ready to roll"); 
    });

})();

```

Works.

```ts
import Koa, { Context } from 'koa';
import Router from 'koa-router';
import { PrismaClient } from '@prisma/client';


const HOST = "127.0.0.1";
const PORT = process.env.PORT || 7000;

const router = new Router();
export const app = new Koa();
router.get('/', (ctx) => ctx.body = { message: 'Welcome to MikroORM Koa TS example, try CRUD on /author and /book endpoints!' });


(async () => {
    app.use(router.routes())
    app.use(router.allowedMethods())
    console.log("Added Routes");
    app.use((ctx, next) => {
      ctx.status = 404;
      ctx.body = { message: 'No route found' };
    });

    app.on("error", err => {
      console.log(err);
    })

    app.listen(PORT, () => {
      console.log("Ready to roll"); 
    });

})();
```

Some really strange behavior here. If I try running `connect` on prism and running some ops it all works fine.

```ts
import Koa, { Context } from 'koa';
import Router from 'koa-router';
import { PrismaClient } from '@prisma/client';


const HOST = "127.0.0.1";
const PORT = process.env.PORT || 7000;

const router = new Router();
export const app = new Koa();
router.get('/', (ctx) => ctx.body = { message: 'Welcome to MikroORM Koa TS example, try CRUD on /author and /book endpoints!' });

  const prisma = new PrismaClient();
prisma.$connect().then(() => {
  console.log("gg");
})
prisma.tokens.count().then((c) => console.log(c));

(async () => {
    app.use(router.routes())
    app.use(router.allowedMethods())
    console.log("Added Routes");
    app.use((ctx, next) => {
      ctx.status = 404;
      ctx.body = { message: 'No route found' };
    });

    app.on("error", err => {
      console.log(err);
    })

    app.listen(PORT, () => {
      console.log("Ready to roll"); 
    });

})();
```

All of the print statements work but the order is problem. The `ready to roll` is printed before prisma.

TLDR;

When prisma starts it somehow kills the main koa app. So not only does the app no longer work,, even in browser/postman its ` ECONNREFUSED`.

## Prisma information

```schema
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = "postgres://daggy:sex@localhost:5432/postgres"
}

model application {
  uu             String   @id @db.Uuid
  appname        String   @db.VarChar(255)
  appdescription String   @db.VarChar(255)
  appurl         String   @db.VarChar(255)
  appuserid      BigInt   @unique
  approved       Boolean
  premium        Boolean
  created_at     DateTime @default(now()) @db.Timestamptz(6)
  updated_at     DateTime @updatedAt @db.Timestamptz(6)
}

model tokens {
  userid     BigInt   @id
  apikey     String   @unique @db.VarChar(64)
  totaluses  Int
  enhanced   Boolean
  ratelimit  Int
  created_at DateTime @default(now()) @db.Timestamptz(6)
  updated_at DateTime @updatedAt @db.Timestamptz(6)
}


```

## Environment & setup

- OS:  Wisons
- Database:  Postgresql
- Node.js version:  14.15.4
- Prisma version: 

```
prisma               : 2.21.1
@prisma/client       : 2.21.1
Current platform     : windows
Query Engine         : query-engine e421996c87d5f3c8f7eeadd502d4ad402c89464d (at node_modules\@prisma\engines\query-engine-windows.exe)
Migration Engine     : migration-engine-cli e421996c87d5f3c8f7eeadd502d4ad402c89464d (at node_modules\@prisma\engines\migration-engine-windows.exe)
Introspection Engine : introspection-core e421996c87d5f3c8f7eeadd502d4ad402c89464d (at node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary        : prisma-fmt e421996c87d5f3c8f7eeadd502d4ad402c89464d (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash : e421996c87d5f3c8f7eeadd502d4ad402c89464d
Studio               : 0.371.0
Done in 3.68s.
```

**Comment by janpio** [comment]
Hmmm, I could run your "Does not Work." example just fine with no modifications. I put my repro repo on Github:  https://github.com/janpio/repro6602 `npm install` and then `npx ts-node index.ts` Where did I go wrong?

**Comment by Daggy1234** [comment]
After trying your example, it worked. I'm trying to figure out the difference.
